serverless/client/client.go

```diff
@@ -386,3 +389,38 @@ func (lst *LogStateTracker) Update(ctx context.Context) error {
 	lst.LatestConsistentRaw, lst.LatestConsistent = cRaw, *c
 	return nil
 }
+
+// CheckConsistency is a wapper function which simplifies verifying consistency between two or more checkpoints.
+func CheckConsistency(ctx context.Context, h hashers.LogHasher, f Fetcher, cp []log.Checkpoint) error {
+	sort.Slice(cp, func(i, j int) bool {
+		return cp[i].Size < cp[j].Size
+	})
+	pb, err := NewProofBuilder(ctx, cp[len(cp)-1], h.HashChildren, f)
+	if err != nil {
+		return fmt.Errorf("failed to create proofbuilder: %v", err)
+	}
+
+	lv := logverifier.New(h)
+
+	// Go through list of checkpoints pairwise, checking consistency
+	var a, b log.Checkpoint
+	for i := 0; i < len(cp)-1; i, a, b = i+1, cp[i], cp[i+1] {
+		if a.Size == b.Size {
+			if bytes.Equal(a.Hash, b.Hash) {
+				continue
+			}
+			return fmt.Errorf("two checkpoints with same size but different hashes")
```

Comment:
Including the hashes & size in this message would be cool.

Reply:
done